### PR TITLE
[Source-MySQL] Enable Enhanced Standard sync with PK -> Cursor switch

### PR DESCRIPTION
**Pull request by nguyenaiden**
## What
Enable PK for initial sync with cursor switch over
[PR](https://github.com/airbytehq/airbyte/pull/30270) with change gated behind flag

### CODE REVIEW

File: airbyte-integrations/connectors/source-mysql/src/test/java/io/airbyte/integrations/source/mysql/MySqlJdbcSourceAcceptanceTest.java [hunk 1]
```diff
@@ -77,25 +107,26 @@ static void init() throws Exception {
...
-        .put(JdbcUtils.DATABASE_KEY, Strings.addRandomSuffix("db", "_", 10))
-        .put(JdbcUtils.USERNAME_KEY, TEST_USER)
-        .put(JdbcUtils.PASSWORD_KEY, TEST_PASSWORD.call())
-        .build());
+                                .put(JdbcUtils.HOST_KEY, container.getHost())
+                                .put(JdbcUtils.PORT_KEY, container.getFirstMappedPort())
+                                .put(JdbcUtils.DATABASE_KEY, Strings.addRandomSuffix("db", "_", 10))
+                                .put(JdbcUtils.USERNAME_KEY, TEST_USER)
+                                .put(JdbcUtils.PASSWORD_KEY, TEST_PASSWORD.call())
+                                .put("standard_via_pk", true)
```
Comment (akashkulk) [inline airbyte-integrations/connectors/source-mysql/src/test/java/io/airbyte/integrations/source/mysql/MySqlJdbcSourceAcceptanceTest.java:115]: Is still required as we are adding this as the default?

File: airbyte-integrations/connectors/source-mysql-strict-encrypt/src/test/java/io/airbyte/integrations/source/mysql_strict_encrypt/MySqlStrictEncryptJdbcSourceAcceptanceTest.java [hunk 2]
```diff
@@ -92,6 +120,7 @@ public void setup() throws Exception {
         .put(JdbcUtils.DATABASE_KEY, Strings.addRandomSuffix("db", "_", 10))
         .put(JdbcUtils.USERNAME_KEY, container.getUsername())
         .put(JdbcUtils.PASSWORD_KEY, container.getPassword())
+        .put("standard_via_pk", true)
```
Comment (akashkulk) [inline airbyte-integrations/connectors/source-mysql-strict-encrypt/src/test/java/io/airbyte/integrations/source/mysql_strict_encrypt/MySqlStrictEncryptJdbcSourceAcceptanceTest.java:123]: Is this still required?

**Review by akashkulk — APPROVED** [review]
Looks good in general - approving to unblock. 
I assume that MySqlPkStrictEncryptJdbcSourceAcceptanceTest.java was renamed to MySqlStrictEncryptJdbcSourceAcceptanceTest.java and the old test deleted (similar with the integration test)

File: airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/initialsync/MySqlFeatureFlags.java
Comment (akashkulk) [inline airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/initialsync/MySqlFeatureFlags.java:1]: I think we can just delete this file for now

File: airbyte-integrations/connectors/source-mysql-strict-encrypt/src/test/java/io/airbyte/integrations/source/mysql_strict_encrypt/MySqlStrictEncryptJdbcSourceAcceptanceTest.java (hunk 2 above)
Comment (nguyenaiden) [reply airbyte-integrations/connectors/source-mysql-strict-encrypt/src/test/java/io/airbyte/integrations/source/mysql_strict_encrypt/MySqlStrictEncryptJdbcSourceAcceptanceTest.java:123]: Oh you right. I don't need that anymore.

**Comment by airbyte-oss-build-runner** [comment]
##  source-mysql test report (commit [`ac7269dc3b`](https://github.com/airbytehq/airbyte/pull/30556/commits/ac7269dc3bc0700be43fff415011ce0817d25d6b)) - ✅

⏲️  Total pipeline duration: 26mn06s 

| Step                                                                | Result   |
|:--------------------------------------------------------------------|:---------|
| Build connector tar                                                 | ✅        |
| Build source-mysql docker image for platform linux/x86_64           | ✅        |
| Java Connector Unit Tests                                           | ✅        |
| Java Connector Integration Tests                                    | ✅        |
| Acceptance tests                                                    | ✅        |
| Validate airbyte-integrations/connectors/source-mysql/metadata.yaml | ✅        |
| Connector version semver check                                      | ✅        |
| Connector version increment check                                   | ✅        |
| QA checks                                                           | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/enable-mysql-pk-sync/1695315148/ac7269dc3bc0700be43fff415011ce0817d25d6b/source-mysql/3.1.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:ac7269dc3bc0700be43fff415011ce0817d25d6b)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-mysql test
```

**Comment by airbyte-oss-build-runner** [comment]
##  source-mysql-strict-encrypt test report (commit [`40007a8dbf`](https://github.com/airbytehq/airbyte/pull/30556/commits/40007a8dbf0d8d03b54d98c83ce96691cadbbc9e)) - ✅

⏲️  Total pipeline duration: 11mn51s 

| Step                                                                               | Result   |
|:-----------------------------------------------------------------------------------|:---------|
| Build connector tar                                                                | ✅        |
| Build source-mysql-strict-encrypt docker image for platform linux/x86_64           | ✅        |
| Java Connector Unit Tests                                                          | ✅        |
| Java Connector Integration Tests                                                   | ✅        |
| Acceptance tests                                                                   | ✅        |
| Validate airbyte-integrations/connectors/source-mysql-strict-encrypt/metadata.yaml | ✅        |
| Connector version semver check                                                     | ✅        |
| Connector version increment check                                                  | ✅        |
| QA checks                                                                          | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/enable-mysql-pk-sync/1695317397/40007a8dbf0d8d03b54d98c83ce96691cadbbc9e/source-mysql-strict-encrypt/3.1.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:40007a8dbf0d8d03b54d98c83ce96691cadbbc9e)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-mysql-strict-encrypt test
```

**Comment by airbyte-oss-build-runner** [comment]
##  source-mysql test report (commit [`40007a8dbf`](https://github.com/airbytehq/airbyte/pull/30556/commits/40007a8dbf0d8d03b54d98c83ce96691cadbbc9e)) - ✅

⏲️  Total pipeline duration: 25mn53s 

| Step                                                                | Result   |
|:--------------------------------------------------------------------|:---------|
| Build connector tar                                                 | ✅        |
| Build source-mysql docker image for platform linux/x86_64           | ✅        |
| Java Connector Unit Tests                                           | ✅        |
| Java Connector Integration Tests                                    | ✅        |
| Acceptance tests                                                    | ✅        |
| Validate airbyte-integrations/connectors/source-mysql/metadata.yaml | ✅        |
| Connector version semver check                                      | ✅        |
| Connector version increment check                                   | ✅        |
| QA checks                                                           | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/enable-mysql-pk-sync/1695317397/40007a8dbf0d8d03b54d98c83ce96691cadbbc9e/source-mysql/3.1.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:40007a8dbf0d8d03b54d98c83ce96691cadbbc9e)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-mysql test
```

**Pull request merged**